cpp/splinepy/splines/nurbs.hpp

```diff
@@ -191,39 +191,50 @@ class Nurbs : public splinepy::splines::SplinepyBase,
     const auto& vector_space = GetWeightedVectorSpace();
 
     // degrees
-    for (std::size_t i{}; i < kParaDim; ++i) {
-      degrees[i] = static_cast<int>(parameter_space.GetDegrees()[i]);
+    if (degrees) {
```

Comment:
I have seen these changes for the other files. What does if (int*) do and what happens in the else case?

Reply:
> In the context of a [direct-initialization](https://en.cppreference.com/w/cpp/language/direct_initialization), a bool object may be initialized from a prvalue of type [std::nullptr_t](https://en.cppreference.com/w/cpp/types/nullptr_t), including nullptr. The resulting value is false. However, this is not considered to be an implicit conversion.

more info [here](https://en.cppreference.com/w/cpp/language/implicit_conversion). This is just to allow this function to act as individual getter for properties as well. Used in degree conformity check. We talked about this offline